fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java

```diff
@@ -629,8 +630,23 @@ private boolean processModifyColumn(ModifyColumnClause alterClause, OlapTable ol
                     // TODO:the case where columnPos is not empty has not been considered
                     if (columnPos == null && col.getDataType() == PrimitiveType.VARCHAR
                             && modColumn.getDataType() == PrimitiveType.VARCHAR) {
-                        col.checkSchemaChangeAllowed(modColumn);
-                        lightSchemaChange = olapTable.getEnableLightSchemaChange();
+                        try {
+                            if (col.getType().isSupportSchemaChangeForCharType(modColumn.getType())) {
```

Comment:
isSupportSchemaChangeForCharType and isSupportSchemaChangeForComplexType only throw ddl exception, then these try and if logic could be removed

Reply:
done!